app/src/main/java/com/example/ithaca_transit_android_v2/NetworkingAdapters.kt

```diff
@@ -44,4 +46,128 @@ class LocationAdapter {
+        @Json(name = "startTime")
+        val startTime: String,
+        @Json(name = "endTime")
+        val endTime: String,
+        @Json(name = "startLocation")
+        val startLocation: JsonLocation,
+        @Json(name = "endLocation")
+        val endLocation: JsonLocation,
+        @Json(name = "stops")
+        val busStops: List<JsonLocation>,
+        @Json(name = "routeNumber")
+        val busNumber: Int
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonLocation(
+        val type: LocationType,
+        val name: String,
+        val lat: Double,
+        val long: Double,
+        val detail: String?
+    )
+
+    @FromJson
+    private fun fromJson(json: DataRoute) : RouteOptions {
+        fun processStringDate(dateString: String): Date {
+            val inputFormatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SS'Z'")
+            return inputFormatter.parse(dateString)
+        }
+
+        fun processStops(locs: List<JsonLocation>): List<Location> {
+            return locs.map{ loc -> Location(LocationType.BUS_STOP, loc.name,
+                Coordinate(loc.lat, loc.long), loc.detail)}
+        }
+
+        fun processDirections(jsonDirection: List<JsonDirection>): List<Direction> {
+            return jsonDirection.map { dir -> Direction(
+                dir.listOfCoordinates.map { c -> Coordinate(c.latitude, c.longitude) },
+                processStringDate(dir.startTime),
+                processStringDate(dir.endTime),
+                Coordinate(dir.startLocation.lat, dir.startLocation.long),
+                Coordinate(dir.endLocation.lat, dir.endLocation.long),
+                processStops(dir.busStops),
+                dir.busNumber
+            )}
+        }
+
+        fun computeBoardInMin(firstBusDirection: Direction): Int {
+            val diff =  firstBusDirection.startTime.time - Calendar.getInstance().time.time
+            return TimeUnit.MINUTES.convert(diff, TimeUnit.MILLISECONDS).toInt()
+        }
+
+        fun processRoutesBus(jsonRoute: List<JsonRoute>): List<Route> {
+            return jsonRoute.map { route -> Route(
+                processDirections(route.directions),
+                Coordinate(route.startLocation.lat, route.startLocation.long),
+                Coordinate(route.endLocation.lat, route.endLocation.long),
+                false,
+                processStringDate(route.arrival),
+                processStringDate(route.depart),
+                computeBoardInMin(processDirections(route.directions)[1])
+                // index [1] is accessed because the second direction is the first bus direction
+            ) }
+        }
+
+        fun processRoutesWalk(jsonRoute: List<JsonRoute>): List<Route> {
+            return jsonRoute.map { route -> Route(
+                processDirections(route.directions),
+                Coordinate(route.startLocation.lat, route.startLocation.long),
+                Coordinate(route.endLocation.lat, route.endLocation.long),
```

Comment:
we should just change the route.startLocation and route.endLocation to Coordinates

Reply:
https://github.com/cuappdev/ithaca-transit-android-v2/pull/21/files/b25a94ab323506c749552b0ce339e2461f22b1ba#r345022551